src/Confluent.Kafka/Impl/SafeKafkaHandle.cs

```diff
@@ -273,6 +273,35 @@ internal WatermarkOffsets GetWatermarkOffsets(string topic, int partition)
             return new WatermarkOffsets(low, high);
         }
 
+        internal IEnumerable<TopicPartitionOffset> OffsetsForTimes(IEnumerable<TopicPartitionTimestamp> timestampsToSearch, int millisecondsTimeout)
+        {
+            var offsets = timestampsToSearch.Select(t => new TopicPartitionOffset(t.TopicPartition, t.Timestamp.UnixTimestampMs)).ToList();
+            IntPtr cOffsets = GetCTopicPartitionList(offsets);
+            try
+            {
+                // The timestamps to query are represented as Offset property in offsets param on input, 
+                // and Offset property will contain the offset on output
+                var errorCode = LibRdKafka.offsets_for_times(handle, cOffsets, (IntPtr) millisecondsTimeout);
+                if (errorCode != ErrorCode.NoError)
+                {
+                    throw new KafkaException(errorCode);
+                }
+
+                var list = GetTopicPartitionOffsetErrorList(cOffsets);
+                var error = list.Where(x => x.Error.Code != ErrorCode.NoError).Select(x => x.Error).FirstOrDefault();
```

Comment:
Oh sorry, we return `TopicPartitionOffset` not `TopicPartitionTimestamp`. So keep the `TopicPartitionTimestamp` type but return `TopicPartitionOffsetError` instead of `TopicPartitionOffset` (and get rid of the check for per partition errors here).